runtime/compiler/codegen/J9TreeEvaluator.cpp

```diff
@@ -595,7 +595,15 @@ uint32_t getInstanceOfOrCheckCastTopProfiledClass(TR::CodeGenerator *cg, TR::Nod
    TR_ByteCodeInfo bcInfo = node->getByteCodeInfo();
    TR_ValueProfileInfoManager *valueProfileInfo = TR_ValueProfileInfoManager::get(comp);
 
-   if (!valueProfileInfo)
+   // We do not have validation record to verify that relocated profiled class
+   // in the load run is instanceof castclass or not. So without that
+   // verification, we could end up generating code where we have a defined
+   // relationship between profiled class and cast class which could not be
+   // true in load run and we could end up with incorrect execution in the
+   // application. 
+   // TODO: Once we have validation record for instanceOfOrCheckCastNoCacheUpdate
+   // enable profiled class test in AOT when SVM is enabled.
+   if (!valueProfileInfo || comp->compileRelocatableCode())
```

Comment:
Do we have an issue backing the TODO and someone to work on it?

Reply:
Opened up just one now: https://github.com/eclipse/openj9/issues/12246 FYI @dsouzai